### PR TITLE
DM-10156: Replace all uses of Calib with PhotoCalib

### PR DESCRIPTION
**Pull request by parejkoj**

### CODE REVIEW

File: python/lsst/afw/image/utils.py [hunk 1]
```diff
@@ -68,23 +67,6 @@ def defineFilter(name, lambdaEff, lambdaMin=np.nan, lambdaMax=np.nan, alias=[],
             Filter.defineAlias(name, a)
 
 
-class CalibNoThrow:
-    """A class intended to be used with python's with statement, to return NaNs for negative fluxes
-    instead of raising exceptions (exceptions may be raised for other purposes).
-
```
Comment (czwa) [inline python/lsst/afw/image/utils.py]: This is the default with `PhotoCalib`, correct?
(Just to make sure I'm understanding the differences correctly.)

File: python/lsst/afw/table/exposure/exposureContinued.py [hunk 2]
```diff
@@ -22,8 +22,21 @@
...
+# class ExposureRecord:
+#     @deprecated(reason="Replaced with getPhotoCalib (will be removed in 18.0)", category=FutureWarning)
+#     def getCalib(self, *args, **kwargs):
+#         return self._getCalib(*args, **kwargs)
+
+#     @deprecated(reason="Replaced with setPhotoCalib (will be removed in 18.0)", category=FutureWarning)
+#     def setCalib(self, *args, **kwargs):
+#         return self._setCalib(*args, **kwargs)
+
+
```
Comment (czwa) [inline python/lsst/afw/table/exposure/exposureContinued.py]: As before, for reference can you note the ticket for this implementation?

File: src/image/ExposureInfo.cc [hunk 3]
```diff
@@ -235,8 +237,8 @@ ExposureInfo::FitsWriteData ExposureInfo::_startWriteFits(lsst::geom::Point2I co
     /*
      * We need to define these keywords properly! XXX
      */
-    data.metadata->set("FLUXMAG0", getCalib()->getFluxMag0().first);
-    data.metadata->set("FLUXMAG0ERR", getCalib()->getFluxMag0().second);
+    // data.metadata->set("FLUXMAG0", getPhotoCalib()->getFluxMag0().first);
+    // data.metadata->set("FLUXMAG0ERR", getPhotoCalib()->getFluxMag0().second);
```
Comment (czwa) [inline src/image/ExposureInfo.cc]: Will these be defined properly?  I assume nothing currently depends on these values?

File: python/lsst/afw/image/utils.py (hunk 1 above)
Comment (parejkoj) [reply python/lsst/afw/image/utils.py]: Yes: `PhotoCalib` never throws due to negative fluxes.

File: src/image/ExposureInfo.cc (hunk 3 above)
Comment (parejkoj) [reply src/image/ExposureInfo.cc]: Oh yeah, those. They can and should be deleted. The comment there is very long standing, and now irrelevant.

**Pull request merged**